### PR TITLE
Disable input parsing

### LINKED ISSUE #1600: multiline entry get scrambled
### Summary

I'm having a weird behavior when using gopass for inserting multiline content containing curly brackets.
For some reason, the secret saved during insertion produces a different output when shown.

This could be related to my setup but do gopass edit/parse what's being inserted in some way that could
produce below changes?

### Steps To Reproduce

start inserting a new entry
```bash
$ echo $EDITOR                   
vim
$ gopass insert -m test
```

Insert the following:
```
test1
test2
{
  "Creator": "the creator"
}
```

try to read back the entry:
```bash
$ gopass show test
test1
"Creator": "the creator"
test2
{
}
```

### Expected behavior

Entry content is exactly the same as inserted.

### Environment

- OS: linux
- gopass Version: 1.10.1

### LINKED ISSUE #1602: add a config option to disable entry parsing
### Summary

I'd like to be able to freely edit entries using gopass and not get parts of the entered text discarded or re-ordered (#1600, #1601). Please provide a config option to disable any kind of parsing on the content of an entry (for `insert -m` or `edit`).

### LINKED ISSUE #1614: Usernames which are numbers are pasted in scientific notation
### Summary

if I have a "user" component in the gpg file, and it's a number, the gopass brige in windows pastes it in scientific 

### Steps To Reproduce

Example, if I have
```
user: 043513
```
it gets pasted in the field in firefox as "4.3513e+4"
### Expected behavior

I expet to see "043513"
### Environment

Windows 10
gopass version 1.9.2
go version (as stated in ```gopass -v``` : 1.14.2

### Edit with second part of the bug

If I click the show details button on gopass brige the leading zeros in a username get omited

### PR DESCRIPTION
**Pull request by AnomalRoil**
Fixes #1602
Fixes #1614
Fixes #1600

RELEASE_NOTES=[BUGFIX] Disabling all kind of parsing on the input
RELEASE_NOTES=[ENHANCEMENT] Adding the flag show -a to disable output parsing and the option parsing to disable all parsing on yaml and KV

### CODE REVIEW

**Comment by AnomalRoil** [comment]
I'm afraid I ended up also taking a stab at #1594 and #1601. I know I should probably have split it into another PR, but I got carried away.

I also replace `-a/-all` with `-n/--noparsing` in the end since it's clearer.

**Comment by AnomalRoil** [comment]
I still need to add regression tests actually.

File: docs/commands/show.md [hunk 1]
```diff
@@ -15,30 +15,93 @@ $ gopass show entry --password
...
 `--password` | `-o` | Display only the password. For use in scripts. Takes precedence over other flags.
 `--revision` | `-r` | Display a specific revision of the entry. Use an exact version identifier from `gopass history` or the special `-N` syntax. Does not work with native (e.g. git) refs.
+`--noparsing` | `-n` | Do not parse the content, disable YAML and Key-Value functions.
 
 ## Details
 
 This section describes the expected behaviour of the `show` command with respect to different combinations of flags and config options.
 
-Note: This section desribed the expected behaviour, not necessarily the observed behaviour. If you notice any discrepancies please file a bug and we will try to fix it.
+Note: This section described the expected behaviour, not necessarily the observed behaviour. If you notice any discrepancies please file a bug and we will try to fix it.
```
Comment (dominikschulz) [inline docs/commands/show.md:36]: Actually that should be `describes`.

File: docs/commands/show.md [hunk 2]
```diff
@@ -15,30 +15,93 @@ $ gopass show entry --password
...
+    password: ******
+    where: the first line is the password
+    
+    
+    and maybe we have a body text
+    below it
+    ```
+
+
+ - the YAML type which implements full YAML support, which means that secrets are parsed as per YAML standard.
```
Comment (dominikschulz) [inline docs/commands/show.md:87]: I wouldn't guarantee `full` YAML support.

File: internal/action/generate.go [hunk 3]
```diff
@@ -287,7 +287,7 @@ func (s *Action) generateSetPassword(ctx context.Context, name, key, password st
 	}
 
 	if content, found := s.renderTemplate(ctx, name, []byte(password)); found {
-		nSec, err := secparse.Parse(content)
+		nSec, err := secparse.Parse(content) // TODO: should we really parse templates input?
```
Comment (dominikschulz) [inline internal/action/generate.go:290]: No, we shouldn't parse here.

File: internal/action/insert.go [hunk 4]
```diff
@@ -130,7 +130,7 @@ func (s *Action) insertSingle(ctx context.Context, name, pw string, kvps map[str
 		sec = gs
 	} else {
 		if content, found := s.renderTemplate(ctx, name, []byte(pw)); found {
-			nSec, err := secparse.Parse(content)
+			nSec, err := secparse.Parse(content) // TODO: should we really parse templates input?
```
Comment (dominikschulz) [inline internal/action/insert.go:133]: no

File: internal/action/show.go [hunk 5]
```diff
@@ -232,7 +243,8 @@ func isUnsafeKey(key string, sec gopass.Secret) bool {
 }
 
 func randAsterisk() string {
-	return strings.Repeat("*", 5+rand.Intn(5))
+	// we could also have a random number of asterisks but testing becomes painful
+	return strings.Repeat("*", 5)
```
Comment (dominikschulz) [inline internal/action/show.go:247]: Good idea.

**Review by dominikschulz — COMMENTED** [review]
Looks good so far, but I'll need to test these changes locally.

File: internal/config/config.go [hunk 6]
```diff
@@ -44,6 +45,7 @@ func New() *Config {
 		ExportKeys:    true,
 		Mounts:        make(map[string]string),
 		Notifications: true,
+		Parsing:       true,
```
Comment (dominikschulz) [inline internal/config/config.go:48]: This needs to be added to all legacy configs, too (see `legacy.go`). Otherwise people will surprised when updating.

Comment (AnomalRoil) [reply internal/config/config.go:48]: Oh, I would have completely missed it. Done.

**Review by dominikschulz — APPROVED** [review]
LGTM

**Pull request merged**